File: app/src/main/java/ru/samtakoy/core/database/room/entities/other/LearnCourseHelper.java
package ru.samtakoy.core.database.room.entities.other;

import java.util.List;

import ru.samtakoy.core.database.room.entities.LearnCourseEntity;


public class LearnCourseHelper {

    public static Long[] getLearnCourseIds(List<LearnCourseEntity> list) {
        Long[] result = new Long[list.size()];
        int i = 0;
        for (LearnCourseEntity learnCourse : list) {
            result[i++] = learnCourse.getId();
        }
        return result;
    }

}

File: app/src/main/java/ru/samtakoy/features/learncourse/data/utils/LearnCourseHelper.kt
package ru.samtakoy.features.learncourse.data.utils

import ru.samtakoy.features.learncourse.domain.model.LearnCourse

object LearnCourseHelper {
    fun getLearnCourseIds(list: List<LearnCourse>): Array<Long> {
        return list.map { it.id }.toTypedArray()
    }
}